lint/registry.go

```diff
@@ -0,0 +1,88 @@
+package lint
+
+import (
+	"fmt"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+// MakeRegistryFromAllFiles creates a *protoregistry.Files with all dependencies present in descs resolved.
+// Missing dependencies will be filled with placeholders. If there are cyclic dependencies, an error will
+// be returned.
+func MakeRegistryFromAllFiles(descs []*descriptorpb.FileDescriptorProto) (*protoregistry.Files, error) {
+	filesSet, err := makeFilesSet(descs)
+	if err != nil {
+		return nil, err
+	}
+
+	f := files{
+		reg:      new(protoregistry.Files),
+		filesSet: filesSet,
+	}
+
+	for _, desc := range descs {
+		if err := f.register(desc.GetName()); err != nil {
+			return nil, err
+		}
+	}
+
+	return f.reg, nil
+}
+
+type files struct {
+	reg *protoregistry.Files
+
+	filesSet map[string]*entry
+}
+
+type entry struct {
+	descProto   *descriptorpb.FileDescriptorProto
+	desc        protoreflect.FileDescriptor
+	registering bool
+}
+
+func (f *files) register(path string) error {
+	e, ok := f.filesSet[path]
+
+	// if it's already been registered (from another import) do nothing. if it doesn't exist,
```

Comment:
Remove comments here, instead, create a function `e.isRegistered()` that returns true if `e.desc != nil`, i.e., codes should speak itself.

Reply:
I'm not a fan of this level of indirection. Adding more functions takes up more space and makes it harder for the reader to click through what's happening. A function like `isRegistered` seems obvious to us, but a reader will still have to click through to understand what that means. I will remove the comment if you think this code is obvious enough.

Reply:
I think the explanation that a placeholder will be filled is useful.